Created Output Log File: /workdir/tasks_run/data/nfb_logs/p004_calculator_script_20240912_14h07m29s.txtexperimenter screen width: 3390
experimenter screen height: 1080
Second monitor resolution: 1920x1080
TO SHOW INSTRUCTIONS, PLEASE PRESS 'r'.
Showing Instructions. Task will start when 's' is pressed (scanner key maps to s on mac, so starting the experiment will trigger the task)
Running Main Calculation Script ... 
Starting Block1 ... 
========================================
Starting Block1, Trial 1... 
========================================
Waiting For New File ...
New File Found In Dir...
Using DICOM:/workdir/tasks_run/data/sambashare/test_dir/95907927929285857075.dcm
Mean ROI Activation: 186.99771118164062
Error:
expected str, bytes or os.PathLike object, not Nifti1Image
Traceback (most recent call last):
  File "/workdir/tasks_run/scripts/ScriptManager.py", line 28, in wrapper
    updated_dictionary = func(*args, **kwargs)
  File "/workdir/tasks_run/scripts/nf_calc_MW.py", line 26, in run_trial
    dictionary = Calculator.get_resid(dictionary=dictionary, block=block, trial=trial)
  File "/workdir/tasks_run/scripts/Calculator.py", line 78, in get_resid
    subprocess.run(['gzip', '-c', nii_img, '>', nii_gz_img])  # compress image to .nii.gz
  File "/usr/local/lib/python3.10/subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/local/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/lib/python3.10/subprocess.py", line 1780, in _execute_child
    self.pid = _posixsubprocess.fork_exec(
TypeError: expected str, bytes or os.PathLike object, not Nifti1Image

Retries left: 3
Using DICOM:/workdir/tasks_run/data/sambashare/test_dir/82735067070452218944.dcm
Mean ROI Activation: 186.99771118164062
Error:
expected str, bytes or os.PathLike object, not Nifti1Image
Traceback (most recent call last):
  File "/workdir/tasks_run/scripts/ScriptManager.py", line 28, in wrapper
    updated_dictionary = func(*args, **kwargs)
  File "/workdir/tasks_run/scripts/nf_calc_MW.py", line 26, in run_trial
    dictionary = Calculator.get_resid(dictionary=dictionary, block=block, trial=trial)
  File "/workdir/tasks_run/scripts/Calculator.py", line 78, in get_resid
    subprocess.run(['gzip', '-c', nii_img, '>', nii_gz_img])  # compress image to .nii.gz
  File "/usr/local/lib/python3.10/subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/local/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/lib/python3.10/subprocess.py", line 1780, in _execute_child
    self.pid = _posixsubprocess.fork_exec(
TypeError: expected str, bytes or os.PathLike object, not Nifti1Image

Getting New Dicom Instead...
Waiting For New File ...
New File Found In Dir...
Retries left: 2
Using DICOM:/workdir/tasks_run/data/sambashare/test_dir/82735067070452218944.dcm
Mean ROI Activation: 186.99771118164062
Error:
expected str, bytes or os.PathLike object, not Nifti1Image
Traceback (most recent call last):
  File "/workdir/tasks_run/scripts/ScriptManager.py", line 28, in wrapper
    updated_dictionary = func(*args, **kwargs)
  File "/workdir/tasks_run/scripts/nf_calc_MW.py", line 26, in run_trial
    dictionary = Calculator.get_resid(dictionary=dictionary, block=block, trial=trial)
  File "/workdir/tasks_run/scripts/Calculator.py", line 78, in get_resid
    subprocess.run(['gzip', '-c', nii_img, '>', nii_gz_img])  # compress image to .nii.gz
  File "/usr/local/lib/python3.10/subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/local/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/lib/python3.10/subprocess.py", line 1780, in _execute_child
    self.pid = _posixsubprocess.fork_exec(
TypeError: expected str, bytes or os.PathLike object, not Nifti1Image

Getting New Dicom Instead...
Waiting For New File ...
New File Found In Dir...
Retries left: 1
Using DICOM:/workdir/tasks_run/data/sambashare/test_dir/17369317810766909646.dcm
Mean ROI Activation: 186.99771118164062
Error:
expected str, bytes or os.PathLike object, not Nifti1Image
Traceback (most recent call last):
  File "/workdir/tasks_run/scripts/ScriptManager.py", line 28, in wrapper
    updated_dictionary = func(*args, **kwargs)
  File "/workdir/tasks_run/scripts/nf_calc_MW.py", line 26, in run_trial
    dictionary = Calculator.get_resid(dictionary=dictionary, block=block, trial=trial)
  File "/workdir/tasks_run/scripts/Calculator.py", line 78, in get_resid
    subprocess.run(['gzip', '-c', nii_img, '>', nii_gz_img])  # compress image to .nii.gz
  File "/usr/local/lib/python3.10/subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/local/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/lib/python3.10/subprocess.py", line 1780, in _execute_child
    self.pid = _posixsubprocess.fork_exec(
TypeError: expected str, bytes or os.PathLike object, not Nifti1Image

Getting New Dicom Instead...
Waiting For New File ...
---- Keyboard Interrupt Detected ----
What Would You Like to Do?
(1) Continue With The Block
(2) End The Session
(3) Start New Block
Ok, Let's End the Session...
Nifti Outdir Cleared
Ending Session Due to: keyboard interrupt
SUBJECT IS DONE. DISPLAYING EXIT MESSAGE FOR 5
dicom dir using: /workdir/tasks_run/data/sambashare/test_dir
